src/test_l3_deepSleepMgr.c

```diff
@@ -0,0 +1,252 @@
+
+   UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+
+/**
+* @brief Trigger deepsleep from the HAL Deepsleep Manager Module
+*
+* This test provides a scope to trigger deepsleep with a timeout.
+*
+* **Test Group ID:** 03@n
+*
+* **Test Case ID:** 002@n
+*
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+void test_l3_deepsleep_manager_hal_Trigger_Deepsleep(void)
+{
+   gTestID = 2;
+   DeepSleep_Return_Status_t status = DEEPSLEEPMGR_SUCCESS;
+
+   UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+   int deep_sleep_timeout;
+   bool isGPIOWakeup = false;
+   bool networkStandby = false;
+
+    // Step 1: Get the Deep Sleep Timeout
+   UT_LOG_MENU_INFO("----------------------------------------------------------");
+   UT_LOG_MENU_INFO("Specify the Deep Sleep Timeout in seconds between 0 and 604800. 0 for no timeout.  ");
+   UT_LOG_MENU_INFO("----------------------------------------------------------");
+   UT_LOG_MENU_INFO("Select Enable/Disable Wakeup Source: ");
+   scanf("%d", &deep_sleep_timeout);
+   readAndDiscardRestOfLine(stdin);
+   if(deep_sleep_timeout < 0 || deep_sleep_timeout > 604800 )
+        {
+            UT_LOG_ERROR("Invalid timeout value");
+            goto exit;
+        }
+    
+   // Step 2: Call PLAT_DS_SetDeepSleep()
+   UT_LOG_INFO("Calling PLAT_DS_SetDeepSleep()");
+   status = PLAT_DS_SetDeepSleep(deep_sleep_timeout, &isGPIOWakeup, networkStandby);
+   UT_LOG_INFO("Result PLAT_DS_SetDeepSleep: DeepSleep_Return_Status_t:[%s]",
+                UT_Control_GetMapString(DeepSleep_Return_Status_mapTable, status));
+
+  exit:
+   UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+
+/**
+* @brief Termination of the HAL Deepsleep Manager Module
+*
+* This test provides a scope to close the HAL Deepsleep Manager module.
+
+*
+* **Test Group ID:** 03@n
+*
+* **Test Case ID:** 003@n
+*
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+
+void test_l3_deepsleep_manager_hal_Term(void)
+{
+   gTestID = 3;
+   DeepSleep_Return_Status_t status = DEEPSLEEPMGR_SUCCESS;
+
+   UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+   // Step 1: Call PLAT_DS_TERM()
+   UT_LOG_INFO("Calling PLAT_DS_TERM()");
+   status = PLAT_DS_TERM();
+   UT_LOG_INFO("Result PLAT_DS_TERM: DeepSleep_Return_Status_t:[%s]",
+                UT_Control_GetMapString(DeepSleep_Return_Status_mapTable, status));
+
+   UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+
+static UT_test_suite_t * pSuite = NULL;
+
+/**
+ * @brief Register the main tests for this module
+ *
+ * @return int - 0 on success, otherwise failure
+ */
+int test_l3_deepSleepMgr_register(void)
+{
+    // Create the test suite
+    pSuite = UT_add_suite("[L3 deepsleep Functions] ", NULL, NULL);
```

Comment:
please assert if pSuite != UT_KVP_STATUS_SUCCESS , instead of returning -1 , since it is a fatal.

Reply:
@ishthiyaqahmed Do you want me to update the L1 and L2 tests with this assert aswell, since they do not have it at the moment.

Reply:
will focus on L3 now.

Reply:
Sure, then the changes have been made here.